WeRadiate-ThermoSense.ino

```diff
@@ -335,43 +355,82 @@ Returns:
+        pinMode(D11, OUTPUT);
+        digitalWrite(D11, HIGH);
+        
+        bool fCompostTemp = checkCompostSensorPresent();
+        
+        if(!fCompostTemp)
+                {
+                gCatena.SafePrintf("No one-wire temperature sensor detected\n");
+                }
+        else
+                {
+                gCatena.SafePrintf("One-wire temperature sensor detected\n");
+                }
+        }
 
  static bool checkCompostSensorPresent(void)
- {
-  
-   sensor_CompostTemp.begin();
-   return sensor_CompostTemp.getDeviceCount() != 0;
-  }
+        {
+        sensor_CompostTemp.begin();
+        return sensor_CompostTemp.getDeviceCount() != 0;
+        }
 
 void setup_built_in_sensors(void)
         {
-          uint32_t flags;
-if (flags & CatenaStm32::fHasBme280)
-    {
-    if (gBme.begin(BME280_ADDRESS, Adafruit_BME280::OPERATING_MODE::Sleep))
-      {
-      fBme = true;
-        gCatena.SafePrintf("BME280 found\n");
-      }
-    else
-      {
-      fBme = false;
-      gCatena.SafePrintf("No BME280 found: check wiring\n");
-      }
-    }
-  else
-    {
-    fBme = false;
-    gCatena.SafePrintf("No BME280 found: check wiring. Just nothing. \n");
-    }
+        uint32_t flags;
+        flags = gCatena.GetPlatformFlags();
+  
+        /* initialize the lux sensor */
+        if (flags & CatenaStm32::fHasLuxSi1113)
+                {
+                if (gSi1133.begin())
+                        {
+                        fLux = true;
+                        gSi1133.configure(0, CATENA_SI1133_MODE_SmallIR);
+                        gSi1133.configure(1, CATENA_SI1133_MODE_White);
+                        gSi1133.configure(2, CATENA_SI1133_MODE_UV);
+                        gSi1133.start();
+                        }
+                else
+                        {
+                        fLux = false;
+                        gCatena.SafePrintf("No Si1133 found: check wiring\n");
+                        }
+                }
+        else
+                {
+                gCatena.SafePrintf("No Si1133 wiring\n");
+                fLux = false;
+                }
+    
+        /* initialize the BME280 */
+        if (flags & CatenaStm32::fHasBme280)
+                {
+                if (gBme.begin(BME280_ADDRESS, Adafruit_BME280::OPERATING_MODE::Sleep))
+                        {
+                        fBme = true;
+                        gCatena.SafePrintf("BME280 found\n");
+                        }
+                else
+                        {
+                        fBme = false;
+                        gCatena.SafePrintf("No BME280 found: check wiring\n");
+                        }
+                }
+        else
+                {
+                fBme = false;
+                gCatena.SafePrintf("No BME280 found: check wiring. Just nothing. \n");
+                }
+
+        if (fLux)
+                {
+                delay(510); /* default measure interval is 500ms */
```

Comment:
I don't think this belongs here; should get timestamp and then later check the light. delays() burn battery power, in any case.

Reply:
should I just remove this since we are not measuring lux anyways?

Reply:
Yes.